docs/source/guides/repository.mdx

```diff
@@ -0,0 +1,213 @@
+
+```py
+>>> from huggingface_hub import create_repo
+>>> create_repo("lysandre/test-dataset", repo_type="dataset")
+'https://huggingface.co/datasets/lysandre/test-dataset'
+```
+
+When you create a repository, you can set your repository visibility with the `private` parameter. For example, if you want to create a private repository:
+
+```py
+>>> from huggingface_hub import create_repo
+>>> create_repo("lysandre/test-private", private=True)
+```
+
+If you want to change the repository visibility at a later time, you can use the [`update_repo_visibility`] function.
+
+### Delete a repository
+
+Delete a repository with [`delete_repo`]. Make sure you want to delete a repository because this is an irreversible process!
+
+Specify the `repo_id` of the repository you want to delete:
+
+```py
+>>> delete_repo(repo_id="lysandre/my-corrupted-dataset", repo_type="dataset")
+```
+
+## Upload and download files
+
+Now that you have created your repository, you are interested in pushing changes to it and downloading files from it.
+
+These 2 topics deserve their own guides. Please refer to the [upload](./upload) and the [download](./download) guides
+to learn how to use your repository.
+
+
+## Branches and tags
+
+Git repositories often make use of branches to store different versions of a same repository.
+Tags can also be use to flag a specific state of your repository, for example when releasing a version.
+More generally, branches and tags are referred as [git references](https://git-scm.com/book/en/v2/Git-Internals-Git-References).
+
+### Create branches and tags
+
+You can create new branch and tags using [`create_branch`] and [`create_tag`]:
+
+```py
+>>> from huggingface_hub import create_branch, create_tag
+
+# Create a branch on a Space repo from `main` branch
+>>> create_branch("Matthijs/speecht5-tts-demo", repo_type="space", branch="handle-dog-speaker")
+
+# Create a tag on a Dataset repo from `v0.1-release` branch
+>>> create_branch("bigcode/the-stack", repo_type="dataset", revision="v0.1-release", tag="v0.1.1", tag_message="Bump release version.")
+```
+
+See also [`delete_branch`] and [`delete_tag`].
+
+### List all branches and tags
+
+You can also list the existing git refs from a repository using [`list_repo_refs`]:
+
+```py
+>>> from huggingface_hub import list_repo_refs
+>>> api.list_repo_refs("bigcode/the-stack", repo_type="dataset")
+GitRefs(
+   branches=[
+         GitRefInfo(name='main', ref='refs/heads/main', target_commit='18edc1591d9ce72aa82f56c4431b3c969b210ae3'),
+         GitRefInfo(name='v1.1.a1', ref='refs/heads/v1.1.a1', target_commit='f9826b862d1567f3822d3d25649b0d6d22ace714')
+   ],
+   converts=[],
+   tags=[
+         GitRefInfo(name='v1.0', ref='refs/tags/v1.0', target_commit='c37a8cd1e382064d8aced5e05543c5f7753834da')
+   ]
+)
+``` 
+
+## Change repository configuration
```

Comment:
I'd also add a sentence or two after this empty section header to describe what its about